include/hip/hcc_detail/functional_grid_launch.hpp

```diff
@@ -160,10 +151,15 @@ void hipLaunchKernelGGLImpl(
     auto it = functions(agent).find(function_address);
 
     if (it == functions(agent).cend()) {
+        int devId{0};
+        char devName[64]{};
+        hipGetDevice(&devId);
```

Comment:
Having HIP APIs invoking other HIP APIs is something that needs to be avoided. There are some instances of it happening in the HIP runtime which need to be cleaned up. But adding more mess is not the way to go.

Reply:
Thanks, I'll do a small change then.